File: Golang/Sum of Unique Elements.go
//1
func sumOfUnique(nums []int) int {
    count := make(map[int]int, len(nums))
    for _, n := range nums {
        count[n]++
    }
    res := 0
    for k, v := range count {
        if v == 1 {
            res += k
        }
    }
    return res
}

//2
func sumOfUnique(nums []int) int {
    unE := make(map[int]int)
    sum := 0
    for _, num := range nums {
        if unE[num] == 0 {
            sum += num
        } else if unE[num] == 1 {
            sum -= num
        }
        unE[num] += 1
    }
    return sum
}

File: Python3/Sum of Unique Elements.py
#1
class Solution:
    def sumOfUnique(self, nums: List[int]) -> int:
        count = {}
        for n in nums:
            if n in count:
                count[n]+=1
            else:
                count[n]=1
        res = 0
        for k, v in count.items():
            if v == 1:
                res += k
        
        return res
        

#2
class Solution:
    def sumOfUnique(self, nums: List[int]) -> int:
        D, s = defaultdict(int), 0
        for n in nums:
            if D[n] == 0:
                s += n
            elif D[n] == 1:
                s -= n
            D[n] += 1
        return s